sprox/sa/provider.py

```diff
@@ -647,7 +647,10 @@ def update(self, entity, params, omit_fields=None):
     # This is hard to test because of some kind of rollback issue in the test framework
     def delete(self, entity, params):  # pragma: no cover
         obj = self._get_obj(entity, params)
-        self.session.delete(obj)
+        try:
+            self.session.delete(obj)
+        except UnmappedInstanceError:
```

Comment:
See previous comment, `UnmappedInstanceError` is again raised due to trying to delete `None`. We should just check for None as `UnmappedInstanceError` hides various other cases and possible bugs which are unrelated to fact that the object existed or not.

Also checking `is not None` better conveys the reason of the check to a reader.